src/dll/Hooks/LoadScripts.cpp

```diff
@@ -0,0 +1,72 @@
+#include "LoadScripts.hpp"
+#include "Addresses.hpp"
+#include "App.hpp"
+#include "Hook.hpp"
+#include "Systems/ScriptCompilationSystem.hpp"
+#include "stdafx.hpp"
+
+namespace
+{
+bool isAttached = false;
+
+bool _CBaseEngine_LoadScripts(RED4ext::CBaseEngine* aEngine, const RED4ext::CString& aPath, uint64_t aTimestamp, uint64_t a4);
+Hook<decltype(&_CBaseEngine_LoadScripts)> CBaseEngine_LoadScripts(Addresses::CBaseEngine_LoadScripts,
+                                                                  &_CBaseEngine_LoadScripts);
+
+bool _CBaseEngine_LoadScripts(RED4ext::CBaseEngine* aEngine, const RED4ext::CString& aPath, uint64_t aTimestamp, uint64_t a4)
+{
+    auto scriptCompilationSystem = App::Get()->GetScriptCompilationSystem();
+    if (scriptCompilationSystem->HasScriptsBlob())
+    {
+        RED4ext::CString scriptsBlobPath = scriptCompilationSystem->GetScriptsBlob().string();
+        return CBaseEngine_LoadScripts(aEngine, scriptsBlobPath, aTimestamp, a4);
```

Comment:
What do you think about:
```suggestion
    auto scriptsBlobPath = scriptCompilationSystem->GetScriptsBlob();
    if (scriptsBlobPath.empty()) // I remembered this had an operator but couldn't find any docs, maybe I dream about it in the past.
    {
        return CBaseEngine_LoadScripts(aEngine, scriptsBlobPath.string(), aTimestamp, a4);
```